enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,206 @@
+        - Certain config changes that the MCO watches don't trigger reboots, but it remains unknown if “moving to layered update path by default” can promise this too.
+    - A: 
+        - Everything rebootless should not be in the image, unless we have live image changes. It is better for those changes to be in writable directories on the disk. Need to keep in mind the priority order of /usr vs /etc.
+        - MCO would have to apply its own internal configs first then the MachineConfigs.
+        - [Actionable item] Get a list of files managed by the MCO to further refine this.
+        - In the future, see if bootc can handle this instead based on the decision and implementation around https://github.com/containers/bootc/issues/76.
+
+## Test Plan
+
+* Phase 1: Please refer to On Cluster Layering enhancement for more details  
+* Phase 2: TBD 
+* Phase 3 : Bootc Switch Over 
+    - Develop an e2e test suite and run payloads with bootc: Tests should target the TestOSImageURL, TestKernelArguments, TestKernelType and TestExtensions. Results will be used to discuss whether bootc is ready to perform rpm-ostree’s tasks.
+    - Create a bootc experimentation sandbox by having the BootcNodeManagement FG: When the FG is on, the rpm-ostree commands will be replaced by equivalent bootc commands to manually test the capability of bootc in non-layered machine configuration workflow. Switch back to rpm-ostree as the default option when not compatible. This experimentation prepares for Scenario 1 in the following section. 
+    - Test bootc with On Cluster Layering: When the BootcNodeManagement and OnClusterBuild FG is on, On Cluster Layering will be the default path for machine configuration, accompanied by bootc which is responsible for image roll out. This test prepares for Scenario 2 in the following section.   
+
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+There are three cases we should consider and try to support under tech preview
+  * Scenario 1: We don’t have layered update path as the default, but we start to provide rpm-ostree + bootc dual support (likely in 4.18)
+  * Scenario 2: We now have layered update path as the default and we still have rpm-ostree + bootc dual support (likely in 4.19)
+  * Scenario 3: We have completed the switch over to layered update path + bootc for image management + dnf for package management (likely in  4.20)
+
+### Tech Preview -> GA
+
+* Unit and E2E test coverage for the two criteria mentioned above 
+* Unit and E2E test coverage for the two criteria mentioned above 
+* The GA of this enhancement is also dependent on:
+    - Bootc GA
+    - On Cluster Layering GA 
+    - Move to Layered Update Path by Default GA
+    - Split RHCOS into Layers & Image Compose 
+    - Move to dnf Interface for RHCOS Package Install 
+    - Local Package Layering on Bootc-based RHCOS 
+* The GA of this enhancement will work with a dual-stream RHCOS 
+
+### Removing a deprecated feature
+
+None
```

Comment:
If all goes well, we should be able to remove the rpm-ostree path at some point in the future. Worth calling that out I think.

Reply:
fixed